### ISSUE TITLE
Support mac/ios native context textbox menu to support writing tools

### ISSUE THREAD

**Issue by BAndysc** [AvaloniaUI/Avalonia#17384]
### Is your feature request related to a problem? Please describe.

Avalonia is not using native controls, and this allows for great flexibility and compatibility between different platforms. However, in some places, native UI is useful, i.e. file dialog pickers - there is nothing more annoying than custom file dialogs, so it is great that Avalonia uses the native API here.

I believe TextBox is another place where native look would be very handy. Especially with the introduction of [writing tools](https://support.apple.com/en-is/guide/mac-help/mchldcd6c260/mac). This Apple-specific feature is very useful, and supporting it in Avalonia would provide a consistent user experience.

![image](https://github.com/user-attachments/assets/ad6ddadc-0ae6-47d1-b7bd-1fcebbdf0822)

### Describe the solution you'd like

Ideally, we would just display the native context menu in textbox. However, programmer custom menus are a problem. Maybe just like with the `NativeMenu`, there could be a `NativeContextMenu` to allow native context menu customization?

### Additional context

According to https://developer.apple.com/videos/play/wwdc2024/10168/, native context menu with writing tools **in custom text boxes** can be achieved using `NSServicesMenuRequestor`.

**Comment by robloo** [comment]
This should be done like IME support I think.

I also don't think we should have the FULL textbox context menu supported. Just an ability to launch some platform-specific tools. In areas like this. However, having a `NativeTextBoxContextMenu` that devs optionally may use per-platform would likely also be fine.

In areas like this for framework consistency we generally have to support the least-common denominator of features across all platforms.

**Comment by BAndysc** [comment]
The more I think about the issue, the more I think this issue could be generalized to just support native context menu.

Especially on platforms like Android and iOS. There already is NativeMenuItem for global menus and tray menu. This could be extended to NativeContextMenu.

**Comment by robloo** [comment]
The NativeContextMenu in the image above has A LOT of functionality. I would be surprised if we can actually fully integrate Avalonia to support it in a reasonable way. Would probably end up with shadowing native TextBoxes or something behind the scenes which gets ugly pretty quickly.

With NativeMenu we basically only support the least common denominator of functionality across all platforms. We just map to native controls like Maui. However, for a native TextBox context menu it would be much more involved I think. It's essentially going the other direction.

Bottom line: I'm hesitant this can be done in a cross-platform, maintainable way.

**Comment by maxkatz6** [comment]
ContextFlyout is not tied to templated popups, and we can have both TextCommandBarFlyout and NativeTextCommandBarFlyout variations with extra menus - https://github.com/AvaloniaUI/Avalonia/discussions/8581.
